USERGUIDE.md

```diff
@@ -2435,13 +2443,143 @@ For VS Code users, the `"erase flash"` [build task](#build-tasks) can be used to
+
+##### SPI Init Function
+
+The input parameters for the `MXC_SPI_Init(...)` function were updated in SPI v2 to allow for more user-selectable options during initialization. This should not cause any errors or behavioral differences with the `MXC_SPI_Init(...)` function when switching between SPI v1 and SPI v2 builds as the input parameters were essentially re-mapped to more descriptive names.
+
+SPI v1:
+
+    :::C
+    int MXC_SPI_Init(mxc_spi_regs_t *spi, 
+                    int masterMode, 
+                    int quadModeUsed, 
+                    int numSlaves, 
+                    unsigned ssPolarity, 
+                    unsigned int hz, 
+                    mxc_spi_pins_t pins)
+
+SPI v2:
+
+    :::C
+    int MXC_SPI_Init(mxc_spi_regs_t *spi, 
+                    mxc_spi_type_t controller_target, 
+                    mxc_spi_interface_t if_mode, 
+                    int numTargets, 
+                    uint8_t ts_active_pol_mask, 
+                    uint32_t freq, 
+                    mxc_spi_pins_t pins)
+
+Input Parameters:
+
+- `mxc_spi_regs_t *spi` remains unchanged.
+- `int masterMode` -> `mxc_spi_type_t controller_target`. The enum `mxc_spi_type_t` was added for increased code readability.
+- `int quadModeUsed` -> `mxc_spi_interface_t if_mode`. Previously, the `MXC_SPI_Init(...)` function could only select between standard (4wire) and quad interface modes. With SPI v2, the user can select either standard (`MXC_SPI_INTERFACE_STANDARD`), quad (`MXC_SPI_INTERFACE_QUAD`), 3wire (`MXC_SPI_INTERFACE_3WIRE`), or dual (`MXC_SPI_INTERFACE_DUAL`) mode.
+- `int numSlaves` -> `int numTargets`. SPI v2 does not use this parameter and was kept to continue supporting SPI v1.
+- `unsigned ssPolarity` -> `uint8_t ts_active_pol_mask`. Updated to more descriptive name.
+- `unsigned int hz` -> `uint32_t freq`.
+- `mxc_spi_pins_t pins` remains unchanged.
+
+##### SPI Config Function
+
+The `int MXC_SPI_Config(mxc_spi_cfg_t cfg)` function was added to reduce the number of helper function calls to set the appropriate settings that the `MXC_SPI_Init(...)` function did not set.
+
+This function also sets up the DMA and acquires DMA TX/RX channels for SPI DMA transactions.
+
+`mxc_spi_cfg_t` struct:
+
+- `mxc_spi_regs_t *spi` - Select SPI Instance to configure.
+- `mxc_spi_clkmode_t clk_mode` - Select clock mode.
+- `uint8_t frame_size` - Select single frame size (2 - 16 bits).
+- `bool use_dma_tx` - Enable SPI DMA TX (acquire and configure TX channel).
+- `bool use_dma_rx` - Enable SPI DMA RX (acquire and configure RX channel).
+- `mxc_dma_regs_t *dma` - Select DMA Instance to configure for SPI DMA (Valid only if `use_dma_tx` or `use_dma_rx` is set to true).
+
+##### SPI Transaction Functions
+
+The SPI v2 Libraries follows the terms used in the user guide: Controller and Target instead of Master and Slave, respectively.
```

Comment:
Suggest changing this to "The SPI naming convention has changed in V2 to match the terminology used by the microcontroller User Guides.  The term "Controller" is now used instead of "Master", and "Target" is used instead of "Slave".